packages/ui/forms/recover-account/index.tsx

```diff
@@ -164,10 +164,10 @@ class RecoverAccountForm extends Component<Props, State> {
                 onBlur={(): void => this.checkMnemonicLength()}
               />
             </View>
-            { mnemonicLengthError && <Text style={style.warningText}>{mnemonicLengthError}</Text> }
-            <Button round fat style={style.submitButton} onPress={() => this.setPIN()} text={recoverAccount} />
-            <Button alternate small arrow style={style.submitButton} onPress={() => this.cancel()} text={cancel} />
           </KeyboardAvoidingView>
+          { mnemonicLengthError && <Text style={style.warningText}>{mnemonicLengthError}</Text> }
+          <Button round fat style={style.submitButton} onPress={this.setPIN.bind(this)} text={recoverAccount} />
```

Comment:
This has the same problem as before; it's an inline function computed with each render.
(this is a **very** common mistake; it's so easy to make and so pervasive that there's a lot of discussion on the topic and even Babel plugins to work around it: [reference](https://www.google.com/search?q=React+inline+function+as+props&oq=React+inline+function+as+props&aqs=chrome..69i57.2960j0j7&sourceid=chrome&ie=UTF-8))

It's actually a very straight-forward issue, and once you understand *why* it's a problem, it becomes very easy to prevent.

**Core concepts:**

 1. React, by default, will not re-render a component if its props (and state) have not changed (via a strict equality check (`===`)). Internalizing this concept is key to writing performant and idiomatic React components.
 2. Strict equality checks only match on primitives (i.e: strings, numbers, object references). So: `{a:123} === {a:123}` is `false` even though the objects have the same shape, and `(() => true) === (() => true)` is `false` even though the two functions have the same implementation… because the left-hand side and the right-hand side are *different object references*. Internalizing this concept is key to preventing props or state from changing unnecessarily.